crates/oxc_diagnostics/src/service.rs

```diff
@@ -73,6 +83,10 @@ impl DiagnosticService {
         self.max_warnings.map_or(false, |max_warnings| self.warnings_count.get() > max_warnings)
     }
 
+    pub fn violated_deny_warnings(&self) -> bool {
+        self.deny_warnings && self.warnings_count.get() > 0
```

Comment:
Does diagnostic service care about this logic 🤔 Does it make sense to move this logic to `CliResult`? Given `CliResult` also has access to `number_of_warnings` and `deny_warwnings`.

Reply:
Yeah good call – I was just following what we do with `quiet` but now I see that's useful within the service.

Reply:
Updated in https://github.com/oxc-project/oxc/pull/1452/commits/29134b9b53836b450d8f5320f473e8d080e7e3d2